src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -134,10 +135,10 @@ def get_version(self):
 
     def get_metadata(self):
         from glob import glob
-        if not extension_exists(self.name):
-            return None
         metadata = {}
         ext_dir = self.path or get_extension_path(self.name)
+        if not ext_dir or not os.path.isdir(ext_dir):
```

Comment:
create extension dir if not exists?

Reply:
The logic to create extension dir if not exists is only added when installing extensions. Here we may not want that.